Example/Tests/DSProviderTransactionsTests.m

```diff
@@ -179,8 +180,13 @@ - (void)testCollateralProviderRegistrationTransaction {
 
     XCTAssertEqual(providerRegistrationTransaction.port, providerRegistrationTransactionFromMessage.port, @"Provider transaction port doesn't match up");
 
-    XCTAssertEqualObjects(providerRegistrationTransaction.inputs, providerRegistrationTransactionFromMessage.inputs, @"Provider transaction inputs are having an issue");
-
+    for (NSUInteger i = 0; i < providerRegistrationTransaction.inputs.count; i++) {
+        DSTransactionInput *input = providerRegistrationTransaction.inputs[i];
+        DSTransactionInput *inputFromMessage = providerRegistrationTransactionFromMessage.inputs[i];
+        XCTAssertTrue(uint256_eq(input.inputHash, inputFromMessage.inputHash), @"Provider transaction input hashes are having an issue");
+        XCTAssertEqual(input.index, inputFromMessage.index, @"Provider transaction input indices are having an issue");
+        XCTAssertEqual(input.sequence, inputFromMessage.sequence, @"Provider transaction input sequences are having an issue");
+    }
```

Comment:
Weird actually... do signatures not match?

Reply:
Actually transaction created from message doesn’t contain input scripts. So comparison of whole inputs is failing

Reply:
I would just do `[providerRegistrationTransactionFromMessage setInputAddress:inputAddress0 atIndex:0];` instead, looks cleaner in my opinion.